File: src/main/java/com/kishlaly/ta/model/indicators/Keltner.java
package com.kishlaly.ta.model.indicators;

import com.kishlaly.ta.model.AbstractModel;

public class Keltner extends AbstractModel {

    private double low;
    private double middle;
    private double top;

    public Keltner(final Long timestamp, final double low, final double middle, final double top) {
        super(timestamp);
        this.low = low;
        this.middle = middle;
        this.top = top;
    }

    public double getLow() {
        return this.low;
    }

    public double getMiddle() {
        return this.middle;
    }

    public double getTop() {
        return this.top;
    }

    public Long getTimestamp() {
        return this.timestamp;
    }

    public boolean valuesPresent() {
        return !Double.isNaN(low) && !Double.isNaN(middle) && !Double.isNaN(top);
    }

}

File: src/main/kotlin/com/kishlaly/ta/model/indicators/Keltner.kt
package com.kishlaly.ta.model.indicators

import com.kishlaly.ta.model.AbstractModel

class Keltner(
    timestamp: Long,
    val low: Double,
    val middle: Double,
    val top: Double
) : AbstractModel(timestamp) {

    override fun valuesPresent() = low != Double.NaN && middle != Double.NaN && top != Double.NaN

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Keltner

        if (low != other.low) return false
        if (middle != other.middle) return false
        if (top != other.top) return false

        return true
    }

    override fun hashCode(): Int {
        var result = low.hashCode()
        result = 31 * result + middle.hashCode()
        result = 31 * result + top.hashCode()
        return result
    }

}